src/app/leverage/provider.tsx

```diff
@@ -142,15 +152,33 @@ export function LeverageProvider(props: { children: any }) {
     setLeverageType(type)
   }
 
+  const onSelectCurrencyToken = (tokenSymbol: string) => {
+    const token = currencyTokens.filter(
+      (token) => token.symbol === tokenSymbol,
+    )[0]
+    if (!token) return
+    setInputToken(token)
+  }
+
+  const onSelectIndexToken = (tokenSymbol: string) => {
+    const token = indexTokens.filter((token) => token.symbol === tokenSymbol)[0]
+    if (!token) return
+    setOutputToken(token)
```

Comment:
Will add more logic here later to take into account `isBuying`. Same for above selection function.